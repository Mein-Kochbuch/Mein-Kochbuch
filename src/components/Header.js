import React from "react";
import {StyleSheet, Text, View} from "react-native";

export default function Header({title}) {

    return (
        <View style={styles.container}>
            <Text style={styles.heading}>
                {title}
            </Text>
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        marginVertical: 12,
        display: "flex",
        justifyContent: "center",
    },
    heading: {
        textAlign: "center",
        fontSize: 50,
    }
})
