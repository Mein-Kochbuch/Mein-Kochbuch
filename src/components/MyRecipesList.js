import React from "react";
import {View} from "react-native";
import MultiListItem from "./MultiListItem";
import CookbookListItem from "./CookbookListItem";

export default function MyRecipesList({cookbooks}) {

    return (
        <View>
            <MultiListItem title={"All Recipes"} icon="all"/>
            <MultiListItem title={"Favorites"} icon="heart"/>
            <View style={{margin: 10}}/>
            {
                cookbooks.map((cookbook) => <CookbookListItem key={cookbook.pk} cookbook={cookbook}/>)
            }
        </View>
    )
}
