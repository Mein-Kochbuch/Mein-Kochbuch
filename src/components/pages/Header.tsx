import React from "react";
import {StyleSheet, Text, View} from "react-native";

interface HeaderProps {
    title: string
}

export default function Header({title}: HeaderProps) {

    return (
        <View style={styles.container}>
            <Text style={styles.heading}>
                {title}
            </Text>
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        marginVertical: 12,
        display: "flex",
        justifyContent: "center",
    },
    heading: {
        textAlign: "center",
        fontSize: 50,
    }
})
