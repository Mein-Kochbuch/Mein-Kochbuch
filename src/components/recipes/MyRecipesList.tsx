import React from "react";
import {View} from "react-native";
import MultiListItem from "../list/MultiListItem";
import CookbookListItem from "../cookbooks/CookbookListItem";
import styled from "styled-components/native";
import {Cookbook} from "../../models/Cookbook";

interface MyRecipeListProps {
    cookbooks: Cookbook[]
}

export default function MyRecipesList({cookbooks}: MyRecipeListProps) {

    return (
        <View>
            <MultiListItem title={"All Recipes"} icon="all"/>
            <MultiListItem title={"Favorites"} icon="heart"/>
            <Spacer/>
            {
                cookbooks.map((cookbook) => <CookbookListItem key={cookbook.pk} cookbook={cookbook}/>)
            }
        </View>
    )
}

const Spacer = styled.View`
  margin: 12px;
`
