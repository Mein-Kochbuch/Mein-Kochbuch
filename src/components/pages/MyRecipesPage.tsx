import React from 'react';
import Header from './Header';
import {ScrollView, StyleSheet} from 'react-native';
import MyRecipesList from '../recipes/MyRecipesList';
import {Cookbook} from '../../models/Cookbook';

export type MyRecipesPageProps = {
  cookbooks: Cookbook[];
};

export default function MyRecipesPage({cookbooks}: MyRecipesPageProps) {
  return (
    <ScrollView style={styles.container}>
      <Header title={'My Recipes'} />
      <MyRecipesList cookbooks={cookbooks} />
    </ScrollView>
  );
}

const styles = StyleSheet.create({
  container: {
    height: '100%',
  },
});
