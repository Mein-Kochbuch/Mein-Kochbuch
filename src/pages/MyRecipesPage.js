import React from "react";
import Header from "../components/Header";
import {ScrollView, StyleSheet} from "react-native";
import MyRecipesList from "../components/MyRecipesList";
import useCookbooks from "../hooks/useCookbooks";

export default function MyRecipesPage() {
    const {cookbooks} = useCookbooks()

    return (
        <ScrollView style={styles.container}>
            <Header title={"My Recipes"}/>
            <MyRecipesList cookbooks={cookbooks}/>
        </ScrollView>
    );
}

const styles = StyleSheet.create({
    container: {
        height: "100%",
    },
})
