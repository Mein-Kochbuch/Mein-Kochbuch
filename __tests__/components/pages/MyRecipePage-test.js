import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import MyRecipesPage from "../../../src/components/pages/MyRecipesPage";
import MyRecipesList from "../../../src/components/recipes/MyRecipesList";
import Header from "../../../src/components/pages/Header"

const cookbooks = [{cookbook: "1", pk: "1"}, {cookbook: "2", pk: "2"}]

jest.mock("../../../src/hooks/useCookbooks", () => {
    return () => {
        return {cookbooks}
    }
});

jest.mock("../../../src/components/recipes/MyRecipesList", () => {
    return (props) => {
        return <div {...props}/>
    }
});

jest.mock("../../../src/components/pages/Header", () => {
    return (props) => {
        return <div {...props}/>
    }
});


it('MyRecipeTest Test', () => {

    const component = renderer.create(
        <MyRecipesPage/>
    );

    expect(component.toJSON()).toMatchSnapshot()
    expect(component.toJSON().children.length).toBe(1)
    expect(component.toJSON().children[0].children.length).toBe(2)
    const testInstance = component.root
    expect(testInstance.findByType(Header).props.title).toBe("My Recipes")
    expect(testInstance.findByType(MyRecipesList).props.cookbooks).toBe(cookbooks)
})
