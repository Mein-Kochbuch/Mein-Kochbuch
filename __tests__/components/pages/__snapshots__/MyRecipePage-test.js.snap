// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyRecipeTest Test 1`] = `
<RCTScrollView
  style={
    Object {
      "height": "100%",
    }
  }
>
  <View>
    <div
      title="My Recipes"
    />
    <div
      cookbooks={
        Array [
          Object {
            "cookbook": "1",
            "id": "1",
          },
          Object {
            "cookbook": "2",
            "id": "2",
          },
        ]
      }
    />
  </View>
</RCTScrollView>
`;
